blog/_posts/2021-07-08-portforwarding-is-awesome.md

```diff
@@ -0,0 +1,292 @@
+---
+slug: "portforwarding-is-awesome"
+date: 2021-07-08
+author: nick
+layout: blog
+title: "Portforwarding should be a tool in every dev toolbox"
+subtitle: "An overview of socat, kubectl port-forward, and how Tilt manages portforwards"
+description: "An overview of socat, kubectl port-forward, and how Tilt manages portforwards"
+image: "/assets/images/portforwarding-is-awesome/tunnel.jpg"
+image_caption: "The Joralemon Street Tunnel, via <a href='https://commons.wikimedia.org/wiki/File:Joralemon_Street_Tunnel_postcard,_1913.jpg'>commons.wikipedia.org</a>."
+tags:
+  - api
+  - networking
+---
+
+In the Unix mindset, every program reads and writes files. So it's important to have a toolkit of basic
+tools like `cat` for working with files!
+
+In 2021, we mostly write servers. That's why it's important to have a toolkit for working with 
+network sockets!
+
+In this post, we're going to do a quick tour of the tools we (Tilt-team) work with everyday,
```

Comment:
```suggestion
In this post, we're going to do a quick tour of the tools we (the Tilt team) work with everyday,
```
Can we have something here to answer the question, "how will I, the reader, become more awesome by reading this post?" It's not clear to me what people are expected to gain here.

Reply:
Yeah, I think there's a natural progression here of 1. `socat` is the bomb but it's the assembly of socket tools 2. `kubectl port-forward` is not ideal for part of daily workflow 3. Tilt is more ergonomic + more suited for dev loop

I like your line later:
> We're not building this API only for internal Tilt use. Tilt exposes them
> publicly, to help connect new tools that build on top of them.

I wonder if it makes sense to allude to that here somehow in the vein of "Tilt does this well AND with the new APIs you don't have to buy in to completely switching your local dev to Tilt to be able to benefit from it"

Reply:
ya, i agree the intro is weak. i took some language from Milas' comment and added some mission statements. wdyt?